contrib/platform/src/com/sun/jna/platform/win32/Advapi32Util.java

```diff
@@ -1695,30 +1673,23 @@ public static void registrySetExpandableStringValue(HKEY root,
     public static void registrySetStringArray(HKEY hKey, String name,
         String[] arr) {
 
-        int charwidth = W32APITypeMapper.DEFAULT == W32APITypeMapper.UNICODE ? Native.WCHAR_SIZE : 1;
-
         int size = 0;
         for (String s : arr) {
-            size += s.length() * charwidth;
-            size += charwidth;
+            size += s.length() + 1; // Null termination
```

Comment:
Nice job with the null but this change is not needed, previous code handled unicode using widestrings.